### PR TITLE
[system test] [doc] LoggingChangeST

### PR DESCRIPTION
**Pull request by see-quick**
### Type of change

- Enhancement / new feature
- Refactoring
- Documentation

### Description

This PR adds a test doc for the `LoggingChangeST` class.

### Checklist

- [x] Make sure all tests pass
- [x] Update documentation

### CODE REVIEW

File: development-docs/systemtests/io.strimzi.systemtest.log.LoggingChangeST.md [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+
+**Description:** This test case checks that changing Logging configuration from internal to external triggers Rolling Update.
+
+**Steps:**
+
+| Step | Action | Result |
+| - | - | - |
+| 1. | Deploy Kafka cluster, without any logging related configuration. | Cluster is deployed. |
+| 2. | Modify Kafka by changing specification of logging to new external value. | Change in logging specification triggers Rolling Update. |
+| 3. | Modify Kafka by changing specification of logging to new logging format. | Change in logging specification triggers Rolling Update. |
```
Comment (henryZrncik) [inline development-docs/systemtests/io.strimzi.systemtest.log.LoggingChangeST.md:28]: We modify only CM in this case not Kafka directly.

File: systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java [hunk 2]
```diff
@@ -467,6 +511,23 @@ && cmdKubeClient().namespace(testStorage.getNamespaceName()).execInPodContainer(
     @ParallelNamespaceTest
     @Tag(BRIDGE)
     @Tag(ROLLING_UPDATE)
+    @TestDoc(
+        description = @Desc("This test dynamically changes the logging levels of the Kafka Bridge and verifies the application behaves correctly with respect to these changes."),
+        steps = {
+            @Step(value = "Configure initial logging levels to OFF.", expected = "Logging levels are set to OFF."),
+            @Step(value = "Deploy Kafka resources asynchronously and wait for their readiness.", expected = "Kafka resources, including the Bridge, become ready."),
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java:518]: I would not say "Kafka resources" when this test checks the Bridge logging. It's a bit confusing when you are reading through it. Also, you are configuring this only in the Bridge, so I would write the steps differently.

File: systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java [hunk 3]
```diff
@@ -685,36 +763,28 @@ void testDynamicallySetClusterOperatorLoggingLevels() {
...
-     *  - rolling-update
-     */
     @ParallelNamespaceTest
     @Tag(ROLLING_UPDATE)
     @Tag(CONNECT)
     @Tag(CONNECT_COMPONENTS)
+    @TestDoc(
+        description = @Desc("This test case verifies dynamic and non-dynamic changes in KafkaConnect logging level."),
+        steps = {
+            @Step(value = "Deploy Kafka cluster and KafkaConnect cluster, the latter with Log level Off.", expected = "Clusters are deployed with KafkaConnect log level set to Off."),
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java:773]:
```suggestion
            @Step(value = "Deploy Kafka cluster and KafkaConnect cluster, the latter with log level Off.", expected = "Clusters are deployed with KafkaConnect log level set to Off."),
```

File: systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java [hunk 4]
```diff
@@ -685,36 +763,28 @@ void testDynamicallySetClusterOperatorLoggingLevels() {
...
+            @Step(value = "Deploy all additional resources, scraper Pod and network policies.", expected = "Additional resources, scraper Pod and network policies are deployed."),
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java:774]: I would also say why the NP are created -> in order to gather the stuff from the Scraper Pod.

File: systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java [hunk 5]
```diff
@@ -1033,6 +1134,20 @@ void testDynamicallySetUnknownKafkaLogger() {
     }
 
     @ParallelNamespaceTest
+    @TestDoc(
+        description = @Desc("Test the dynamic setting of an unknown Kafka logger value without triggering a rolling update."),
+        steps = {
+            @Step(value = "Create test storage and resources with required namespaces and node pools.", expected = "Resources are created with the correct configuration."),
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java:1140]: As before - the information about test storage is not useful for the overall test case. And what is correct configuration? Is there something changed?

File: systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java [hunk 6]
```diff
@@ -1058,6 +1173,20 @@ void testDynamicallySetUnknownKafkaLoggerValue() {
...
 
     @ParallelNamespaceTest
+    @TestDoc(
+        description = @Desc("Test dynamically setting Kafka external logging configurations to ensure logging updates without triggering unnecessary rolling updates, and verifies correct logging behavior."),
+        steps = {
+            @Step(value = "Create a ConfigMap with initial logging settings.", expected = "ConfigMap is created with the 'external-cm' name."),
+            @Step(value = "Apply external logging configuration to Kafka brokers.", expected = "Kafka brokers are configured with the specified external logging."),
+            @Step(value = "Verify no rolling update occurs after initial logging update.", expected = "No rolling update occurs and logging change is verified."),
+            @Step(value = "Update ConfigMap with new logger levels.", expected = "ConfigMap is updated and rolling update is expected."),
+            @Step(value = "Verify rolling update completion and new logging settings applied.", expected = "Rolling update is completed successfully and new settings are active.")
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java:1183]: I guess this is a bit different -> you deploy Kafka with some external CM configuration and then you change it, then you verify that there is no rolling update, right?

File: systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java [hunk 7]
```diff
@@ -1187,6 +1316,22 @@ void testDynamicallySetKafkaExternalLogging() {
     @Tag(ROLLING_UPDATE)
     @Tag(MIRROR_MAKER2)
     @Tag(CONNECT_COMPONENTS)
+    @TestDoc(
+        description = @Desc("Test to dynamically set logging levels for MirrorMaker2 and verify the changes are applied without pod rolling."),
+        steps = {
+            @Step(value = "Set initial logging level to OFF and deploy resources.", expected = "Resources are deployed with logging level set to OFF."),
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java:1322]: What resources are you deploying? Where is the `OFF` set?

File: systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java [hunk 8]
```diff
@@ -1306,6 +1451,21 @@ void testDynamicallySetMM2LoggingLevels() {
     @ParallelNamespaceTest
     @Tag(ROLLING_UPDATE)
     @Tag(MIRROR_MAKER2)
+    @TestDoc(
+        description = @Desc("Test to ensure logging levels in Kafka MirrorMaker2 follow a specific hierarchy."),
+        steps = {
+            @Step(value = "Set up source and target Kafka cluster.", expected = "Kafka clusters are deployed and ready."),
+            @Step(value = "Configure Kafka MirrorMaker2 with initial logging levels.", expected = "Logging configuration is applied with log level 'OFF'."),
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java:1458]:
```suggestion
            @Step(value = "Configure Kafka MirrorMaker2 with initial logging level set to `OFF`.", expected = "Logging configuration is applied with log level 'OFF'."),
```

**Review by henryZrncik — APPROVED** [review]
LGTM Thanks.

**Pull request merged**